src/main/java/ir/sahab/uncaughtexceptionrule/UncaughtExceptionExtension.java

```diff
@@ -0,0 +1,30 @@
+package ir.sahab.uncaughtexceptionrule;
+
+import org.junit.jupiter.api.extension.ExtensionContext;
+import org.junit.jupiter.api.extension.InvocationInterceptor;
+import org.junit.jupiter.api.extension.ReflectiveInvocationContext;
+
+import java.lang.reflect.Method;
+
+/**
+ * A JUnit 5 extension for catching exceptions in threads other than the JUnit main test thread.
+ * Note: you can't use this extension if you're running Concurrent tests.
```

Comment:
Please explain why we also have implemented the InvocationInterceptor (related scenario which we've covered).